pkg/virt-api/webhooks/validating-webhook/admitters/pod-eviction-admitter.go

```diff
@@ -0,0 +1,80 @@
+package admitters
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"k8s.io/api/admission/v1beta1"
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	virtv1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+)
+
+type PodEvictionAdmitter struct {
+	ClusterConfig *virtconfig.ClusterConfig
+	VirtClient    kubecli.KubevirtClient
+}
+
+func (admitter *PodEvictionAdmitter) Admit(ar *v1beta1.AdmissionReview) *v1beta1.AdmissionResponse {
+	if !admitter.ClusterConfig.LiveMigrationEnabled() {
+		return &v1beta1.AdmissionResponse{Allowed: true}
+	}
+
+	launcher, err := admitter.VirtClient.CoreV1().Pods(ar.Request.Namespace).Get(ar.Request.Name, metav1.GetOptions{})
+	if err != nil {
+		// we don't want to block the request because we were not able to find the pod
+		return &v1beta1.AdmissionResponse{Allowed: true}
+	}
+
+	if value, exists := launcher.GetLabels()[virtv1.AppLabel]; !exists || value != "virt-launcher" {
+		// this is not a virt-launcher app so there's no reason to block it
+		return &v1beta1.AdmissionResponse{Allowed: true}
+	}
+
+	domainName, exists := launcher.GetAnnotations()[virtv1.DomainAnnotation]
+	if !exists {
+		// we don't know who is the VMI that created this pod so we will just have to allow this eviction
+		return &v1beta1.AdmissionResponse{Allowed: true}
+	}
+
+	vmi, err := admitter.VirtClient.VirtualMachineInstance(ar.Request.Namespace).Get(domainName, &metav1.GetOptions{})
+	if err != nil {
+		return &v1beta1.AdmissionResponse{Allowed: true}
+	}
+
+	conditions := append(vmi.Status.Conditions, virtv1.VirtualMachineInstanceCondition{
```

Comment:
On a quick check it looks like the conditions is never removed, can that be?
I think that an explicit status subsection where you can log the eviction timepoint and the node on which it happens is better than a condition. I would expect that the status subsection is cleared if the VMI then runs on another node.

Reply:
Heh :) I just thought on this situation a few minutes ago too.
Thanks, I'll fix that.